hack/postrelease/golang/calico_oss_test.go

```diff
@@ -0,0 +1,132 @@
+package calico_oss_test
+
+import (
+	"calico_postrelease/pkg/container"
+	"calico_postrelease/pkg/registry"
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var release_tag = "v3.27.4"
+
+var expected_arches = []string{"amd64", "arm64", "s390x", "ppc64le"}
+
+var expected_images = []string{
+	"calico/node",
+	"calico/ctl",
+	"calico/apiserver",
+	"calico/typha",
+	"calico/cni",
+	"calico/kube-controllers",
+	// "calico/upgrade",
+	"calico/flannel-migration-controller",
+	"calico/dikastes",
+	// "calico/pilot-webhook",
+	"calico/pod2daemon-flexvol",
+	"calico/csi",
+}
+
+// Commenting these out because we don't currently tag the
+// individual windows images. Should we?
+// var expected_windows_tags = []string{
+// 	"windows-ltsc2022",
+// 	"windows-1809",
+// }
+
+var expected_windows_images = []string{
+	"calico/cni-windows",
+	"calico/node-windows",
+}
+
+func TestGolang(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Calico OSS Postrelease Test Suite")
+}
+
+var _ = Describe(
+	"Validate Container Image Publishing",
+	func() {
+		var operatorImage container.Image
+		var quayRegistry registry.QuayRegistry = registry.NewQuayRegistry()
+		var gcrRegistries = []registry.GCRRegistry{
```

Comment:
Rather than having code to deal with the different registries, perhaps consider using a client library like https://github.com/regclient/regclient ?

Reply:
That's a good and extremely obvious idea, I wish I'd considered that earlier. Thanks!